### ISSUE TITLE
astropy 6.0.1 breaks using quantities as unit exponents

### ISSUE THREAD

**Issue by maxnoe** [astropy/astropy#16260]
### Description

This works in astropy 6.0.0:

```
u.m**u.Quantity(2)
```

but raises an exception in astropy 6.0.1:

```
u.m**u.Quantity(2)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[2], line 1
----> 1 u.m**u.Quantity(2)

File ~/.local/conda/envs/gammapy-dev/lib/python3.11/site-packages/astropy/units/core.py:831, in UnitBase.__pow__(self, p)
    829 def __pow__(self, p):
    830     p = validate_power(p)
--> 831     return CompositeUnit(1, [self], [p], _error_check=False)

File ~/.local/conda/envs/gammapy-dev/lib/python3.11/site-packages/astropy/units/core.py:2341, in CompositeUnit.__init__(self, scale, bases, powers, decompose, decompose_bases, _error_check)
   2339         scale *= unit.scale**power
   2340         self._bases = unit.bases
-> 2341         self._powers = [
   2342             sanitize_power(operator.mul(*resolve_fractions(p, power)))
   2343             for p in unit.powers
   2344         ]
   2346     self._scale = sanitize_scale(scale)
   2347 else:
   2348     # Regular case: use inputs as preliminary scale, bases, and powers,
   2349     # then "expand and gather" identical bases, sanitize the scale, &c.

File ~/.local/conda/envs/gammapy-dev/lib/python3.11/site-packages/astropy/units/core.py:2342, in <listcomp>(.0)
   2339         scale *= unit.scale**power
   2340         self._bases = unit.bases
   2341         self._powers = [
-> 2342             sanitize_power(operator.mul(*resolve_fractions(p, power)))
   2343             for p in unit.powers
   2344         ]
   2346     self._scale = sanitize_scale(scale)
   2347 else:
   2348     # Regular case: use inputs as preliminary scale, bases, and powers,
   2349     # then "expand and gather" identical bases, sanitize the scale, &c.

File ~/.local/conda/envs/gammapy-dev/lib/python3.11/site-packages/astropy/units/utils.py:291, in sanitize_power(p)
    288 denom = getattr(p, "denominator", None)
    289 if denom is None:
    290     # This returns either a (simple) Fraction or the same float.
--> 291     p = maybe_simple_fraction(p)
    292     # If still a float, nothing more to be done.
    293     if isinstance(p, float):

File ~/.local/conda/envs/gammapy-dev/lib/python3.11/site-packages/astropy/units/utils.py:217, in maybe_simple_fraction(p, max_denominator)
    215 if p == 0 or p.__class__ is int or p.__class__ is Fraction:
    216     return p
--> 217 n, d = p.as_integer_ratio()
    218 a = n // d
    219 # Normally, start with 0,1 and 1,0; here we have applied first iteration.

File ~/.local/conda/envs/gammapy-dev/lib/python3.11/site-packages/astropy/units/quantity.py:1108, in Quantity.__getattr__(self, attr)
   1103 """
   1104 Quantities are able to directly convert to other units that
   1105 have the same physical type.
   1106 """
   1107 if not self._include_easy_conversion_members:
-> 1108     raise AttributeError(
   1109         f"'{self.__class__.__name__}' object has no '{attr}' member"
   1110     )
   1112 def get_virtual_unit_attribute():
   1113     registry = get_current_unit_registry().registry

AttributeError: 'Quantity' object has no 'as_integer_ratio' member
```

### Expected behavior

Bugfix releases don't introduce breaking changes

### How to Reproduce

```python
import astropy.units as u

u.m**u.Quantity(2)
```

### Versions

astropy 6.0.1

**Comment by maxnoe** [comment]
@pllim

**Comment by neutrinoceros** [comment]
I was going to say that https://github.com/astropy/astropy/pull/16058 was the likely suspect but you've already confirmed it with a bisection. Thank you ! I'll take a look now and see if we can fix it simply or if we should revert #16058 instead.

**Comment by neutrinoceros** [comment]
found a simple fix, will open a PR in a minute

**Comment by neutrinoceros** [comment]
fixed in #16261
